modules/yieldmoBidAdapter.js

```diff
@@ -429,8 +432,13 @@ function openRtbSite(bidRequest, bidderRequest) {
 /**
  * @return Object OpenRTB's 'device' object
  */
-function openRtbDevice() {
+function openRtbDevice(bidRequest) {
+  let ip;
+  if (bidRequest.params && bidRequest.params.device) {
+    ip = bidRequest.params.device.ip;
+  }
   return {
+    ip,
```

Comment:
@ym-abaranov have you tested when `ip = undefined` with our ad server?

Reply:
Yes, I've tested and it works. there is some logic in pb.js which is turning the obj into a string using JSON.stringify(), so that's how `ip: undefined` removes from the obj. I guess it's better to change the logic just to make sure that we are adding `ip` only if present because pb.js might be changed and we can fail with that later on.